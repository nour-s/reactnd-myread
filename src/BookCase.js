import React from "react";
import PropTypes from "prop-types";
import Shelf from "./Shelf";
import { Link } from "react-router-dom";

class BookCase extends React.Component {
	render() {
		return (
			<div className="list-books">
				<div className="list-books-title">
					<h1>MyReads</h1>
				</div>
				<div className="list-books-content">
					<div>
						{this.props.shelfs.map(shelf => (
							<Shelf
								onMoveBook={this.props.onMoveBook}
								key={shelf.id}
								shelf={shelf}
							/>
						))}
					</div>
				</div>
				<div className="open-search">
					{(this.props.showSearchLink || false) && (
						<Link to={{ pathname: "/search" }}>Add a book</Link>
					)}
				</div>
			</div>
		);
	}
}

BookCase.propTypes = {
	shelfs: PropTypes.array
};

export default BookCase;
