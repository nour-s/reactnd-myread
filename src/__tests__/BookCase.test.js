import React from "react";
import renderer from "react-test-renderer";
import BookCase from "../BookCase";
import { MemoryRouter } from "react-router";
import { mount } from "enzyme";

describe("BookCase test", () => {
	let shelfs = [
		{ id: "shelf 1", books: [{ id: 1, title: "Title 1" }] },
		{ id: "shelf 2", books: [{ id: 2, title: "Title 2" }] }
	];
	let rawComponent = (
		<MemoryRouter>
			<BookCase shelfs={shelfs} showSearchLink={true} />
		</MemoryRouter>
	);
	it("BookCase should match snapshot", () => {
		const component = renderer.create(rawComponent);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("BookCase should have a link to the Search page", () => {
		const component = mount(React.createElement(() => rawComponent));
		var link = component.find("Link");
		expect(link.prop("to").pathname).toBe("/search");
	});
});
