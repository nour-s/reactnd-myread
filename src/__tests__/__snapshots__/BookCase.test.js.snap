// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookCase test BookCase should match snapshot 1`] = `
<div
  className="list-books"
>
  <div
    className="list-books-title"
  >
    <h1>
      MyReads
    </h1>
  </div>
  <div
    className="list-books-content"
  >
    <div>
      <div
        className="bookshelf"
      >
        <h2
          className="bookshelf-title"
        />
        <div
          className="bookshelf-books"
        >
          <ol
            className="books-grid"
          >
            <li>
              <div
                className="book"
              >
                <div
                  className="book-top"
                >
                  <div
                    className="book-cover"
                    style={
                      Object {
                        "backgroundImage": "url()",
                        "height": 193,
                        "width": 128,
                      }
                    }
                  />
                  <div
                    className="book-shelf-changer"
                  >
                    <select
                      onChange={[Function]}
                      value="none"
                    >
                      <option
                        disabled={true}
                        value="move"
                      >
                        Move to...
                      </option>
                      <option
                        disabled={false}
                        value="currentlyReading"
                      >
                        Currently Reading
                      </option>
                      <option
                        disabled={false}
                        value="read"
                      >
                        Read
                      </option>
                      <option
                        disabled={false}
                        value="none"
                      >
                        None
                      </option>
                    </select>
                  </div>
                </div>
                <div
                  className="book-title"
                >
                  Title 1
                </div>
              </div>
            </li>
          </ol>
        </div>
      </div>
      <div
        className="bookshelf"
      >
        <h2
          className="bookshelf-title"
        />
        <div
          className="bookshelf-books"
        >
          <ol
            className="books-grid"
          >
            <li>
              <div
                className="book"
              >
                <div
                  className="book-top"
                >
                  <div
                    className="book-cover"
                    style={
                      Object {
                        "backgroundImage": "url()",
                        "height": 193,
                        "width": 128,
                      }
                    }
                  />
                  <div
                    className="book-shelf-changer"
                  >
                    <select
                      onChange={[Function]}
                      value="none"
                    >
                      <option
                        disabled={true}
                        value="move"
                      >
                        Move to...
                      </option>
                      <option
                        disabled={false}
                        value="currentlyReading"
                      >
                        Currently Reading
                      </option>
                      <option
                        disabled={false}
                        value="read"
                      >
                        Read
                      </option>
                      <option
                        disabled={false}
                        value="none"
                      >
                        None
                      </option>
                    </select>
                  </div>
                </div>
                <div
                  className="book-title"
                >
                  Title 2
                </div>
              </div>
            </li>
          </ol>
        </div>
      </div>
    </div>
  </div>
  <div
    className="open-search"
  >
    <a
      href="/search"
      onClick={[Function]}
    >
      Add a book
    </a>
  </div>
</div>
`;
